reducers/plugins/index.js

```diff
@@ -137,9 +170,16 @@ function components(state = {}, action) {
         if (action.name && action.data) {
             const nextState = {...state};
             const currentArray = nextState[action.name] || [];
-            const nextArray = [...currentArray];
+            if (action.name === 'PostDropdownMenu' && action.data.parentMenuId) {
+                const subMenuData = buildSubMenu(currentArray, action.data);
+                if (!subMenuData) {
+                    return state;
+                }
+                action.data = subMenuData;
```

Comment:
can we not update `action.data` and have a var that's like `actionData = action.data` which is updated here?

Reply:
Done.